File: src/main/java/gregtechlite/gtlitecore/api/advancement/AdvancementCriterion.java
package gregtechlite.gtlitecore.api.advancement;

import net.minecraft.advancements.ICriterionInstance;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.ResourceLocation;

public interface AdvancementCriterion extends ICriterionInstance
{

    boolean test(EntityPlayerMP player);

    void setId(ResourceLocation id);

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/advancement/AdvancementCriterion.kt
package gregtechlite.gtlitecore.api.advancement

import net.minecraft.advancements.ICriterionInstance
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.util.ResourceLocation

interface AdvancementCriterion : ICriterionInstance
{

    fun test(player: EntityPlayerMP?): Boolean

    fun setId(id: ResourceLocation)
}